coderd/coderdtest/coderdtest.go

```diff
@@ -73,6 +73,7 @@ type Options struct {
 
 	// IncludeProvisionerD when true means to start an in-memory provisionerD
 	IncludeProvisionerD bool
+	APIBuilder          func(*coderd.Options) *coderd.API
```

Comment:
I think a broad abstraction could lead to unintentional patterns emerging of developers passing an entirely custom set of options.

I'd prefer to make this `Enterprise bool` instead. Or `AGPL` so we default to Enterprise.

Reply:
That'd make the `coderdtest` package incompatible with AGPL, but that isn't shipped anyways, so I don't think it's a big deal.